pyeudiw/satosa/default/openid4vp_backend.py

```diff
@@ -0,0 +1,327 @@
+                        f"Loaded configuration: {json.dumps(config)}")
+        
+    def register_endpoints(self) -> list[tuple[str, Callable[[Context], Response]]]:
+        """
+        Creates a list of all the endpoints this backend module needs to listen to. In this case
+        it's the authentication response from the underlying OP that is redirected from the OP to
+        the proxy.
+        :rtype: Sequence[(str, Callable[[satosa.context.Context], satosa.response.Response]]
+        :return: A list that can be used to map the request to SATOSA to this endpoint.
+        """
+        url_map = []
+        for k, v in self.config['endpoints'].items():
+            url_map.append(
+                (
+                    f"^{self.name}/{v.lstrip('/')}$",
+                    getattr(self, f"{k}_endpoint")
+                )
+            )
+            _endpoint = f"{self.client_id}{v}"
+            self._log_debug(
+                "OpenID4VPBackend",
+                f"Exposing backend entity endpoint = {_endpoint}"
+            )
+            if k == 'get_response':
+                self.registered_get_response_endpoint = _endpoint
+            elif k == 'response':
+                self.absolute_redirect_url = _endpoint
+            elif k == 'request':
+                self.absolute_request_url = _endpoint
+            elif k == "status":
+                self.absolute_status_url = _endpoint
+        return url_map
+    
+    def start_auth(self, context: Context, internal_request) -> Response:
+        """
+        This is the start up function of the backend authorization.
+
+        :type context: satosa.context.Context
+        :type internal_request: satosa.internal.InternalData
+        :rtype satosa.response.Response
+
+        :param context: the request context
+        :param internal_request: Information about the authorization request
+        :return: response
+        """
+        return self.pre_request_endpoint(context, internal_request)
+    
+    def pre_request_endpoint(self, context: Context, internal_request, **kwargs) -> Response:
+        """
+        This endpoint is called by the User-Agent/Wallet Instance before calling the request endpoint.
+        It initializes the session and returns the request_uri to be used by the User-Agent/Wallet Instance.
+
+        :type context: the context of current request
+        :param context: the request context
+        :type internal_request: satosa.internal.InternalData
+        :param internal_request: Information about the authorization request
+
+        :return: a response containing the request_uri
+        :rtype: satosa.response.Response
+        """
+
+        self._log_function_debug(
+            "pre_request_endpoint", context, "internal_request", internal_request)
+
+        session_id = context.state["SESSION_ID"]
+        state = str(uuid.uuid4())
+
+        # TODO: do not init the session if the context is not linked to any
```

Comment:
please update the related issue when this PR will be approved, to match the right line number and facilitate the tracking

Reply:
pelase add a logger.warning by saying that this feature is not implemented yet